scripts/docker/build-panuser-image.py

```diff
@@ -0,0 +1,186 @@
+    parser.add_argument(
+        '--user',
+        dest='user_name',
+        type=str,
+        default=(os.environ.get('PANUSER') or getusername()),
+        help=('Name of the PANUSER in the image. '
+              'If specified, the user must exist.'))
+    parser.add_argument(
+        '--group',
+        dest='group_name',
+        type=str,
+        help=('Primary group of the user PANUSER in the image. '
+              'If specified, the group must exist.'))
+    parser.add_argument(
+        '--pandir',
+        dest='pandir',
+        type=str,
+        default=(os.environ.get('PANDIR') or '/var/panoptes'),
+        help='Root directory of PANOPTES software and data.')
+    parser.add_argument(
+        '--base-image',
+        dest='base_image',
+        type=str,
+        default='ubuntu:18.04',
+        help='Base image from which to create this image.')
+    parser.add_argument(
+        '--command',
+        dest='command',
+        type=str,
+        default='/bin/bash',
+        help='Command to run when the image is launched in a container.')
+    parser.add_argument('--tag', dest='tag', type=str, help='Tag to apply to the created image.')
+    parser.add_argument('--verbose', '-v', action='store_true')
+
+    args = parser.parse_args()
+
+    def arg_error(msg):
+        print(msg, '\n', file=sys.stderr)
+        parser.print_help(file=sys.stderr)
+        sys.exit(1)
+
+    try:
+        pwent = pwd.getpwnam(args.user_name)
+        user_name = pwent.pw_name
+        user_id = pwent.pw_uid
+        group_id = pwent.pw_gid
+    except KeyError:
+        arg_error(f'User with name "{args.user_name}" not found.')
+
+    if args.group_name:
+        try:
+            grent = grp.getgrnam(args.group_name)
+            assert args.group_name == grent.gr_name
+            group_name = grent.gr_name
+            group_id = grent.gr_gid
+        except KeyError:
+            arg_error(f'Group with name "{args.group_name}" not found.')
+    else:
+        # Not putting a try-except here because a KeyError implies that
+        # something is wrong with the pwd and grp lookups, not with the
+        # user's input.
+        grent = grp.getgrgid(group_id)
+        assert group_id == grent.gr_gid
+        group_name = grent.gr_name
+
+    if args.pandir:
+        if not os.path.isabs(args.pandir):
+            arg_error(f'PANOPTES directory must be an absolute path, not: {args.pandir}')
+        pandir = args.pandir
+    else:
+        pandir = None
+
+    dockerfile = generate_dockerfile(
+        base_image=args.base_image,
+        user_name=user_name,
+        user_id=user_id,
+        group_name=group_name,
+        group_id=group_id,
+        pandir=pandir,
+        command=args.command)
+    if args.verbose:
+        print('Using this dockerfile:')
+        print('#' * 80)
+        print(dockerfile)
+        print('#' * 80)
+
+    tag = args.tag or f'panuser-{user_name}'
+
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
The point is to be able to create a docker image that works for the current (user name, uid, group name, gid), where that might be the info for a developer, for the CI account or for the panoptes user on a scope. Recall that user and group panoptes might exist on all scopes, but not with the same uid and gid.

So this script will create a new image from a base image (which could be an image with all of the apt dependencies already installed). I'm currently using it with an ubuntu:18.04 base image so that I can test how well my changes to the install scripts, etc., are working.